src/ui/map.js

```diff
@@ -1266,16 +1293,33 @@ class Map extends Camera {
      * });
      * @see [Example: Create a draggable point](https://docs.mapbox.com/mapbox-gl-js/example/drag-a-point/)
      */
-    off(type: MapEvent, layerId: any, listener: any) {
+    off(type: MapEvent, layerIds: any, listener: any) {
         if (listener === undefined) {
-            return super.off(type, layerId);
+            return super.off(type, layerIds);
+        }
+
+        if (!Array.isArray(layerIds)) {
+            layerIds = [layerIds];
         }
 
+        const areLayerArraysEqual = (arr1, arr2) => {
```

Comment:
Thanks so much for this contribution @omerbn 🙇🏽 , really appreciated! 

I have one main piece of feedback that should get rid of the complexity in this function. Since we don't care about the order of `layerIds` passed into `on()` or `off()` we can use a `Set` to track the `layers <-> listener` mapping.
The `layers` returned from `_createDelegatedListener` can be a `Set` directly. So you'd need to create one temporary `Set` here to check for equality.